joysticks.py

```diff
@@ -1,29 +1,38 @@
 """Joystick-related utilities"""
 
-import pygame
 
 class Joysticks:
     """Provides data about usb joysticks (and other
      controllers pygame recognizes as joysticks) via pygame
     """
-    def __init__(self):
-        pygame.init()
 
-        self._joysticks = [pygame.joystick.Joystick(x) for x in range(pygame.joystick.get_count())]
+    def __init__(self, pygame):
+        """Initialize joysticks"""
+
+        self._pygame = pygame
+
+        self._pygame.init()
+
+        self._scan_joysticks()
+
+    def _scan_joysticks(self):
+        # List of all joysticks plugged in (at time of initialization)
+        self._joysticks = [self._pygame.joystick.Joystick(
+            x) for x in range(self._pygame.joystick.get_count())]
```

Comment:
Nice comprehension. Bit of a nit, but shouldn't we have `Joystick(x)` all together? Having `x` on its own looks a bit too important. ;-)

Reply:
Good point, I didn't catch that.